crates/lsp/src/main_loop.rs

```diff
@@ -156,13 +157,10 @@ pub(crate) struct LspState {
     /// Resolver to look up [`Settings`] given a document [`Url`]
     pub(crate) workspace_settings_resolver: WorkspaceSettingsResolver,
 
-    /// The negociated encoding for document positions. Note that documents are
+    /// The negotiated encoding for document line offsets. Note that documents are
     /// always stored as UTF-8 in Rust Strings. This encoding is only used to
-    /// translate UTF-16 positions sent by the client to UTF-8 ones.
-    pub(crate) position_encoding: PositionEncoding,
-
-    /// The set of tree-sitter document parsers managed by the `GlobalState`.
-    pub(crate) parsers: HashMap<Url, tree_sitter::Parser>,
+    /// translate UTF-16 positions sent by the client to UTF-8 byte offsets.
+    pub(crate) encoding: LineOffsetEncoding,
```

Comment:
We end up translating the `lsp_types::PositionKindEncoding` from the Client to a `LineOffsetEncoding` as fast as possible, and that's what we hold onto and use throughout the LSP.

(`PositionEncoding` was also from biome-lsp-converters and we don't need that anymore)